--- /usr/lib/initcpio/hooks/encrypt	2023-03-21 16:50:52.000000000 +0700
+++ /etc/initcpio/hooks/cirnocrypt	2023-08-18 12:36:30.598339298 +0700
@@ -94,13 +94,22 @@
             fi
             # Ask for a passphrase
             if [ ${dopassphrase} -gt 0 ]; then
+                
+                if cat /cirnocrypt > /dev/null; then
+                    _GREETING=$(cat /cirnocrypt)
+                else
+                    _GREETING="Unable to read greeting! Write ascii greeting to /etc/initcpio/cirnocrypt and run mkinitcpio.conf"
+                fi
+                    
+                _GREETING_PLYMOUTH="$_GREETING"
+
                 if command -v plymouth >/dev/null 2>&1 && plymouth --ping 2>/dev/null; then
                     plymouth ask-for-password \
-                        --prompt="A password is required to access the ${cryptname} volume" \
+                        --prompt="$_GREETING_PLYMOUTH" \
                         --command="cryptsetup open --type luks --key-file=- ${resolved} ${cryptname} ${cryptargs} ${CSQUIET}"
                 else
                     echo ""
-                    echo "A password is required to access the ${cryptname} volume:"
+                    echo "$_GREETING"
 
                     #loop until we get a real password
                     while ! eval cryptsetup open --type luks ${resolved} ${cryptname} ${cryptargs} ${CSQUIET}; do
